includes/class-scaip-block-settings.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Newspack Ads SCAIP Block Settings
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Ads;
+
+use Newspack_Ads\Settings;
+
+/**
+ * Newspack Ads SCAIP Block Settings Class.
+ */
+final class SCAIP_Block_Settings {
+
+	/**
+	 * Initialize settings.
+	 */
+	public static function init() {
+		add_action( 'newspack_ads_settings_list', [ __CLASS__, 'settings_list' ] );
+		add_filter( 'scaip_allowing_insertion_blocks', [ __CLASS__, 'filter_allowed_blocks' ] );
+	}
+
+	/**
+	 * Register SCAIP Block Settings.
+	 *
+	 * @param array $settings_list List of settings.
+	 *
+	 * @return array Updated list of settings.
+	 */
+	public static function settings_list( $settings_list ) {
+		if ( ! defined( 'SCAIP_PLUGIN_FILE' ) ) {
+			return $settings_list;
+		}
+
+		return array_merge(
+			[
+				[
+					'description' => esc_html__( 'SCAIP Block Settings', 'newspack-ads' ),
+					'help'        => esc_html__( 'Configure which block types allow ad insertion within article content.' ),
+					'section'     => 'scaip_blocks',
+					'key'         => 'active',
+					'type'        => 'boolean',
+					'default'     => false,
+				],
+				[
+					'description' => esc_html__( 'Allowed Block Types', 'newspack-ads' ),
+					'help'        => esc_html__( 'Select which block types should allow ad insertion.' ),
```

Comment:
```suggestion
					'help'        => esc_html__( 'Select which block types should be included when calculating ad insertion. Warning: selected blocks are eligible to be followed by an ad.' ),
```

We should probably be clearer about what this means. Not sure if I did a good job with my suggestion, though 😄

Reply:
That's an odd way of putting it. This setting governs which block may be followed by an ad. "calculating ad insertion" sounds like it's about the percentage position of ad placement in a post. 

Updated to: "Select which block types can be followed by an ad.".

Reply:
> "calculating ad insertion" sounds like it's about the percentage position of ad placement in a post.

But it is the case https://github.com/Automattic/super-cool-ad-inserter-plugin/blob/c11a669bf70c53442d2ee4b866fc27223451dd08/inc/scaip-shortcode-inserter.php#L111-L116

Reply:
I see, a different way of looking at it. But WDYT about "Select which block types can be followed by an ad".? IMO that's what it boils down to for the user. By default, only paragraphs can be followed by an ad. This PR adds ability to have more blocks followed by an ad.

Reply:
Right, but on top of that, it also changes the calculation strategy. If I have an ad to be rendered **after 2 blocks** and the following article structure:

```html
<h2>Title</h2>
<p>Paragraph 1</p>
<p>Paragraph 2</p>
```

With default settings, it'll render here:

```html
<h2>Title</h2>
<p>Paragraph 1</p>
<p>Paragraph 2</p>
<!-- AD -->
```

If I include `Heading` in the options, it'll render here:

```html
<h2>Title</h2>
<p>Paragraph 1</p>
<!-- AD -->
<p>Paragraph 2</p>
```

I'm ok with phrasing, I just want to make sure the implications are clear 😄

Reply:
Yeah, I'm worried this is too much of technical context :/